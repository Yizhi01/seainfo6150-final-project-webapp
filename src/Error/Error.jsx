import React from "react";
import { Link } from "react-router-dom";
import errorImage from "../images/errorImage.png";
import styles from './Error.module.css';

const Error = () => {
return (
  <div className={styles.errorContainer}>
    <img className={styles.errorImage} src={errorImage} alt="error" />
    <h1>Oops! Something's missing...</h1>
    <p>It seems like we donut find the page you're looking for. It doesn't exist, isn't available or was loading incorrectly.</p>
    <Link className={styles.backBtn} to="/">Back to Home</Link>
  </div>
)
}

export default Error
