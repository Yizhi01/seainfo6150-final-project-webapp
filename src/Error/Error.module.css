.errorContainer {
  margin: 0px 100px 40px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.errorImage {
  width: 90%;
  margin: 0 0 10px;
}

.errorContainer h1 {
  font: bold 3em 'Open Sans', sans-serif;
  margin: 0 0 10px;
}

.errorContainer p {
  font-size: 1em;
}

.backBtn {
  margin: 30px 0;
  padding: 10px 20px;
  background-color: #222831;
  color: #fff;
  border: 3px solid #222831;
  border-radius: 5px;
  text-decoration: none;
}

.backBtn:hover {
  background-color: #fff;
  color: #222831;
  border: 3px solid #222831;
}