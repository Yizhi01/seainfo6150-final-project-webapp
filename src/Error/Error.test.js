import React from "react";
import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import Error from "./Error.jsx";

it("renders the Error component correctly", () => {
  const { container } = render(
    <Router>
        <Error />
    </Router>
  );
  expect(container).toMatchSnapshot();
});