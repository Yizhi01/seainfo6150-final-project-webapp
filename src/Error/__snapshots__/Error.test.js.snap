// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Error component correctly 1`] = `
<div>
  <div
    class="errorContainer"
  >
    <img
      alt="error"
      class="errorImage"
      src="errorImage.png"
    />
    <h1>
      Oops! Something's missing...
    </h1>
    <p>
      It seems like we donut find the page you're looking for. It doesn't exist, isn't available or was loading incorrectly.
    </p>
    <a
      class="backBtn"
      href="/"
    >
      Back to Home
    </a>
  </div>
</div>
`;
